api/src/main/java/com/github/retrooper/packetevents/util/mappings/VersionedRegistry.java

```diff
@@ -38,6 +38,10 @@ public final class VersionedRegistry<T extends MappedEntity> implements IRegistr
     private final Map<String, T> typeMap = new HashMap<>();
     private final Map<Byte, Map<Integer, T>> typeIdMap = new HashMap<>();
 
+    //Each version registry may have a synchronized registry cache, for convenience and enhanced performance.
+    @Nullable
+    private SimpleRegistry<T> synchedRegistry = null;
```

Comment:
If this cache registry within the VersionedRegistry is available, (not null) then it will be used.